xbmc/utils/SeekHandler.cpp

```diff
@@ -79,16 +160,45 @@ bool CSeekHandler::InProgress() const
 
 void CSeekHandler::Process()
 {
-  if (m_timer.GetElapsedMilliseconds() > time_before_seek)
+  if (m_timer.GetElapsedMilliseconds() > m_seekDelay)
   {
     if (!g_infoManager.m_performingSeek && m_timer.GetElapsedMilliseconds() > time_for_display) // TODO: Why?
       g_infoManager.SetSeeking(false);
+
     if (m_requireSeek)
     {
       g_infoManager.m_performingSeek = true;
+
+      // calculate the seek time
       double time = g_infoManager.GetTotalPlayTime() * m_percent * 0.01;
```

Comment:
I assume this is where our rounding difference (the 1sec) offset comes from. We probably should differ between analogue (percentual seeks) and our seek steps with fixed time values. I don't see why we should calculate the fixed time steps into percentages and back - we probably should do it the other way around and calculate the resulting seconds for analogue seeks in CSeekHandler::Seek(). So drop m_percent and use a m_seconds instead.